driver-sync/src/main/com/mongodb/client/internal/ClientSessionImpl.java

```diff
@@ -102,6 +108,21 @@ public void startTransaction(final TransactionOptions transactionOptions) {
         setPinnedMongosAddress(null);
     }
 
+    private void checkServerVersionForTransactionSupport() {
+        ClusterDescription clusterDescription = delegate.getCluster().getCurrentDescription();
+        int wireVersion = 0;
+        for (ServerDescription serverDescription : clusterDescription.getServerDescriptions()) {
+            if (serverDescription.getType() != ServerType.UNKNOWN) {
+                wireVersion = serverDescription.getMaxWireVersion();
```

Comment:
Should we check all the servers instead of stopping at the first and use the minimum wire version?  The way it is now there is randomness depending on which server description we happen to encounter first.

Reply:
I now loop through all server descriptions to get the lowest available max wire version.